arelle/PluginManager.py

```diff
@@ -493,48 +493,50 @@ def pluginClassMethods(className: str) -> Iterator[Callable[..., Any]]:
         for method in pluginMethodsForClass:
             yield method
 
-def addPluginModule(url):
-    moduleInfo = None
-    if sys.version_info < (3, 10):
-        unloaded_arelle_plugins = [e for e in entry_points().get('arelle.plugin', []) if e.name == url]
-    else:
-        unloaded_arelle_plugins = entry_points(group='arelle.plugin', name=url)
-    if unloaded_arelle_plugins:
-        if len(unloaded_arelle_plugins) != 1:
-            error_msg = f'Multiple pip installed plugins with name {url} in group arelle.plugin'
-            logPluginTrace(error_msg, logging.ERROR)
-        pluginUrl = unloaded_arelle_plugins[0].load()
-        moduleInfo = moduleModuleInfo(pluginUrl())
-    if not moduleInfo or not moduleInfo.get("name"):
-        moduleInfo = moduleModuleInfo(url)
-    if moduleInfo and moduleInfo.get("name"):
-        name = moduleInfo["name"]
-        removePluginModule(name)  # remove any prior entry for this module
-        def _addPluginModule(moduleInfo):
-            _name = moduleInfo.get("name")
-            if _name:
-                # add classes
-                for classMethod in moduleInfo["classMethods"]:
-                    classMethods = pluginConfig["classes"].setdefault(classMethod, [])
-                    _name = moduleInfo["name"]
-                    if _name and _name not in classMethods:
-                        classMethods.append(_name)
-                for importModuleInfo in moduleInfo.get('imports', EMPTYLIST):
-                    _addPluginModule(importModuleInfo)
-                pluginConfig["modules"][_name] = moduleInfo
-        _addPluginModule(moduleInfo)
-        global pluginConfigChanged
-        pluginConfigChanged = True
-        return moduleInfo
-    return None
+
+def addPluginModule(pluginModuleInfo: dict[str, Any]) -> dict[str, Any] | None:
```

Comment:
We probably want to check to make sure that this change wont break wf_arelle [here](https://github.com/Workiva/wf_arelle/blob/master/service/jobs/validation/plugin/plugin_utils.py#L65-L76).